feature-libs/product-configurator/common/shared/utils/common-configurator-utils.service.ts

```diff
@@ -65,18 +68,23 @@ export class CommonConfiguratorUtilsService {
    * @param cart Cart
    * @returns Cart identifier
    */
-  public getCartId(cart: Cart): string {
+  getCartId(cart: Cart): string {
     return cart.user.uid === OCC_USER_ID_ANONYMOUS ? cart.guid : cart.code;
   }
+
   /**
-   * Gets cart user
+   * Gets user assigned to a cart which we can use for OCC requests
    * @param cart Cart
-   * @returns User identifier
+   * @returns Observable of user identifier
    */
-  public getUserId(cart: Cart): string {
-    return cart.user.uid === OCC_USER_ID_ANONYMOUS
-      ? cart.user.uid
-      : OCC_USER_ID_CURRENT;
+  getUserId(cart: Cart): Observable<string> {
+    return this.userIdService
+      .getUserId()
+      .pipe(
+        map((id) =>
+          cart.user.uid === OCC_USER_ID_ANONYMOUS ? cart.user.uid : id
```

Comment:
So far, we used the `user` embedded in the `cart` only for getting `email` in the _guest checkout_ functionality (when there was no real account).

Why we should use `user` embedded in the `cart` object in your functionality, but not only the one from UserIdService?

Reply:
Agree, not needed. We can use userIdService directly and can get rid of this method